components/footer/footer.jsx

```diff
@@ -1,42 +1,86 @@
-import React from 'react';
-import { MofoFooter } from 'mofo-ui';
+import React from "react";
 
-const Footer = () => {
-  const FOOTER_LINKS = [
-    {
-      iconType: `github`,
-      link: `https://github.com/mozilla/network-pulse`,
-      text: `Github`
-    },
-    {
-      iconType: `legal`,
-      link: `https://www.mozilla.org/about/legal/terms/mozilla/`,
-      text: `Legal`
-    },
-    {
-      iconType: `cc-license`,
-      link: `https://creativecommons.org/licenses/by/4.0/`,
-      text: `License`
-    },
-    {
-      iconType: `privacy`,
-      link: `https://www.mozilla.org/privacy/websites/`,
-      text: `Privacy`
-    }
-  ];
-
-  const ORGS = [
-    {
-      name: `mozilla`,
-      link: `https://mozilla.org`,
-      description: (<p className="dark-theme body-small">Mozilla is a global non-profit dedicated to putting you in control of your online experience and shaping the future of the web for the public good. Visit us at <a href="https://mozilla.org">mozilla.org</a>.</p>),
-      className: `mozilla`
-    }
-  ];
-
-  return (
-    <MofoFooter footerLinks={FOOTER_LINKS} orgs={ORGS} />
-  );
-};
-
-export default Footer;
+export default class MofoFooter extends React.Component {
+  renderSocialLinks() {
+    return (
+      <ul className="d-flex pl-0">
+        <li className="mr-2">
+          <a
+            className="twitter small d-inline-block"
+            href="https://twitter.com/mozilla">
+            <span className="sr-only">Twitter</span>
+          </a>
+        </li>
+        <li className="mr-2">
+          <a
+            className="instagram small d-inline-block"
+            href="https://www.instagram.com/mozilla/">
+            <span className="sr-only">Instagram</span>
+          </a>
+        </li>
+        <li className="mr-2">
+          <a
+            className="github small d-inline-block"
+            href="https://www.github.com/mozilla/foundation.mozilla.org/">
+            <span className="sr-only">Github</span>
+          </a>
+        </li>
+      </ul>
+    );
+  }
+  renderFooterLinks() {
```

Comment:
What are the classnames `footer-link-*` for? If they are not being used elsewhere let's remove these classnames. Don't forget we want these links to open in a new tab!

I would simplify repetitive code in this method into something like

```jsx

import React from "react";

// this also makes updating links in the future easier
const FOOTER_LINKS = [
    { text: `Donate`, link: `link to donate` },
    { text: `Legal`, link: `link to legal` }
    ...
    ...
];

export default class MofoFooter extends React.Component {
  ...
  renderFooterLinks() {
    let links = FOOTER_LINKS.map(link => {
        return <li className="col-auto mb-2" key={link.text}>
          <a href={link.link} target="_blank">{link.text}</a>
        </li>;
    });

    return (
        <ul className="link-list list-unstyled mb-0">
          {links}
        </ul>
      );
    }
  }
  ...
}
```

Reply:
let's simplify `renderSocialLinks()` as well

Reply:
Since this is a static component, is there really a benefit to having this be dynamically generated? In trying to initially migrate this to Pulse with Youri, this felt overly complicated and I suggested simply hardcoding the footer, since it's not being pulled in as part of the MoFo UI library anymore.

Reply:
I thought about that too. But if we decide to hardcode this then later when we have to plug in the sign up form (which will be a React component) then we'll have to go with the Foundation site way that is using vanilla JS to target the node and ReactDOM to inject the form which I think might be less straightforward.

Reply:
I'm not sure I follow. We need to trigger the React component somehow, what's the difference between triggering it as it is now or limiting the scope to just the form?

Though my suggestion was that we put the markup for the footer right into the render method, so it's still a React component but without the need to iterate over items and generate the HTML.

Reply:
Ah. I thought you meant putting the static markup as HTML, in a `.html` file and use Foundation site's way to find a DOM node to inject the form portion of it.

If the code is all gonna be in one React component I _think_ putting the "static content" right into the render method vs create helper render method to iterate over links essentially are the same (as it's all JSX we are talking about here). I prefer the latter since it makes the code easier to maintain and increase code legibility (e.g., when we have to update a classname we won't have to update 6 instances of the same wrapper). With the array approach it does add more build time. But it's probably just a small fraction of ms we are talking about here and it's fast enough that we can simply ignore.

I don't have a strong opinion here as footer section is unlikely to be changed often. But I'm hoping we can utilize JS methods more as we are building React components. 😎

Reply:
Yeah, I'm thinking more about engineer time to make the logic work vs the time to copy/paste a piece of markup. Sounds like Youri should go with whichever approach she prefers/is easier?

Reply:
@alanmoo @mmmavis if you guys are ok with it, I'm going to keep the logic as is if there aren't any syntax/logic errors please give this a +1 when you have a chance.